src/services/db/GitFileSystemService.ts

```diff
@@ -200,17 +235,48 @@ export default class GitFileSystemService {
         )
       }
 
-      return ResultAsync.fromPromise(
-        this.git.cwd(`${EFS_VOL_PATH}/${repoName}`).checkout(BRANCH_REF).pull(),
-        (error) => {
-          if (error instanceof GitError) {
-            return new GitFileSystemError(error.message)
-          }
+      return this.ensureCorrectBranch(repoName).andThen(() =>
+        ResultAsync.fromPromise(
+          this.git.cwd(`${EFS_VOL_PATH}/${repoName}`).pull(),
+          (error) => {
+            // Full error message 1: Your configuration specifies to merge
+            // with the ref 'refs/heads/staging' from the remote, but no
+            // such ref was fetched.
+            // Full error message 2: error: cannot lock ref
+            // 'refs/remotes/origin/staging': is at <new sha> but expected <old sha>
+            // Full error message 3: Cannot fast-forward your working tree.
+            // Full error message 4: Need to specify how to reconcile divergent branches.
+            // These are known errors that can be safely ignored
```

Comment:
just to confirm the above errors does not mean git pull has failed?

Will the git pull succeed in updating the local file contents for the above scenarios?

Reply:
These errors mean that _one_ of the git pulls have failed, but overall it would have succeeded in updating the local file contents.

Due to the number of read calls made when loading a page, it is possible that multiple git pulls are running concurrently. The first git pull would have succeeded eventually, as it is the one that has acquired the necessary locks to update the local files. However, it is possible that subsequent git pulls happening at the same time would throw errors (like the 4 here) as they would have "seen" the local directory in an inconsistent state (the first git pull is in progress). Hence, it is safe for us to ignore these errors, as the first git pull would have succeeded.

Reply:
in realtion to: `Full error message 4: Need to specify how to reconcile divergent branches.`
what is our behvaiour when there is a conflict? do we force merge?

Reply:
Hmm there shouldn't be a conflict because based on this MVP, as only reads are being performed. A conflict will only occur if the local copy has additional changes beyond the original HEAD commit that Git cannot recover from.